### PR TITLE
[node] load project tar

### LINKED ISSUE #8: [@subql/node] support load subquery project from its tarball pack
It is promised in the grant, and we have already supported packing subquery project into tarball.

For `@subql/node`, it will need to unpack the tarball file into a temp directory, then it will be no different with current flow.
Also the checksum of the tarball file need to be recorded in db.

### PR DESCRIPTION
**Pull request by jiqiang90**
Closes #8 support unpack the tarball file into a temp directory, and return the temp path for subquery project

### CODE REVIEW

File: packages/node/src/utils/project.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import tar from 'tar';
+
+export async function processPath(projectPath: string): Promise<string> {
```
Comment (ianhe8x) [inline packages/node/src/utils/project.ts:6]: can use a better name, like prepareProjectDir

File: packages/node/src/utils/project.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  const stats = fs.statSync(projectPath);
+  if (stats.isFile()) {
+    const sep = path.basename(projectPath);
```
Comment (ianhe8x) [inline packages/node/src/utils/project.ts:9]: `sep` means seperator, we don't really need the original basename to name temp dir

File: packages/node/src/configure/configure.module.ts [hunk 3]
```diff
@@ -72,10 +72,7 @@ export class ConfigureModule {
         },
         {
           provide: SubqueryProject,
-          useValue: SubqueryProject.create(
-            loadProjectManifest(projectPath),
-            projectPath,
-          ),
+          useFactory: async () => SubqueryProject.create(projectPath),
```
Comment (ianhe8x) [inline packages/node/src/configure/configure.module.ts:75]: if path is not a valid tar file, the application will crash. We need to catch the error to make the reason of crash more clear.

**Review by ianhe8x — CHANGES_REQUESTED** [review]

File: packages/node/src/utils/project.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import fs from 'fs';
```
[the remark this answers is not in the document]
Comment (jiqiang90) [reply packages/node/src/utils/project.ts:4]: license header added

File: packages/node/src/utils/project.ts (hunk 1 above)
Comment (jiqiang90) [reply packages/node/src/utils/project.ts]: Name changed

File: packages/node/src/utils/project.ts (hunk 2 above)
Comment (jiqiang90) [reply packages/node/src/utils/project.ts]: Fixed

File: packages/node/src/configure/configure.module.ts (hunk 3 above)
Comment (jiqiang90) [reply packages/node/src/configure/configure.module.ts]: catch error added

File: packages/node/src/configure/project.model.ts [hunk 5]
```diff
@@ -1,13 +1,21 @@
...
+  ProjectManifest,
+  SubqlDataSource,
+} from '@subql/common';
+import { prepareProjectDir } from '../utils/project';
 
 export class SubqueryProject implements ProjectManifest {
-  static create(source: ProjectManifest, path: string): SubqueryProject {
+  static async create(path: string): Promise<SubqueryProject> {
+    const projectPath = await prepareProjectDir(path);
+    console.log(projectPath);
```
Comment (ianhe8x) [inline packages/node/src/configure/project.model.ts:14]: Is this necessary output? remove if it is only for debugging

**Review by ianhe8x — CHANGES_REQUESTED** [review]

File: packages/node/src/configure/project.model.ts (hunk 5 above)
Comment (jiqiang90) [reply packages/node/src/configure/project.model.ts]: removed

**Review by ianhe8x — APPROVED** [review]

**Pull request merged**